### PR TITLE
guix: autodetect build arch in the Dockerfile

### PR DESCRIPTION
**Pull request by prusnak**
This PR makes Docker autodetect arch during the container build.

It also contains overall improvements of the Dockerfile

### CODE REVIEW

File: guix/Dockerfile [hunk 1]
```diff
@@ -32,17 +31,17 @@ RUN wget -qO- "https://guix.carldong.io/signing-key.pub" | guix archive --author
...
-    useradd -g guixbuild -G guixbuild           \
-            -d /var/empty -s "$(which nologin)" \
-            -c "Guix build user ${i}" --system  \
-          "guixbuilder${i}";                    \
-  done
-
-CMD guix-daemon --build-users-group=guixbuild   \
-                --substitute-urls="https://guix.carldong.io https://ci.guix.gnu.org"
+RUN groupadd --system guixbuild
+RUN for i in $(seq -w 1 10); do                  \
```
Comment (fanquake) [inline guix/Dockerfile:35]: Please leave this as `32`. I've increased it so I can perform guix builds with more than at most 10 builders.

**Review by fanquake — CHANGES_REQUESTED** [review]
@dongcarl did you want to have a quick look here as well?

**Comment by MarcoFalke** [comment]
You could also base this on `debian:bullseye`, which already ships with guix on all archs that guix claims to support as well.

File: guix/Dockerfile (hunk 1 above)
Comment (prusnak) [reply guix/Dockerfile]: Addressed in 5ad2fb8ddcbc07df478126f513ef0f6d303d329c

Comment (dongcarl) [reply guix/Dockerfile]: Hmm, not to spend too much time on this (and definitely don't prevent this from merge), but perhaps `$(nproc)` would be good here? I'm thinking number of builders should probably scale with number of threads?

**Comment by dongcarl** [comment]
Concept ACK! Paves the way for running Guix builds on different architectures! I made the manifest in Bitcoin Core host-architecture agnostic so that we could pursue this in the future!

tenor

**Comment by dongcarl** [comment]
> You could also base this on `debian:bullseye`, which already ships with guix on all archs that guix claims to support as well.

True, perhaps we can do so in another PR!

File: guix/Dockerfile (hunk 1 above)
Comment (MarcoFalke) [reply guix/Dockerfile]: Heh, is there a reason to not set this to 256? :sweat_smile:

Comment (prusnak) [reply guix/Dockerfile]: @dongcarl Good idea, but the machine building the container can have different `$(nproc)` than the target machine using the container.

**Comment by prusnak** [comment]
I think it's better not to drag Debian into the mix and keep it simple

File: guix/Dockerfile (hunk 1 above)
Comment (dongcarl) [reply guix/Dockerfile]: > @dongcarl Good idea, but the machine building the container can have different `$(nproc)` than the target machine using the container.
Oh true... Let's just make it a build arg then with a default of 32?

> Heh, is there a reason to not set this to 256? 😅
I would think the scheduler might be very unhappy with us if we set this to 256 hehe

**Comment by fanquake** [comment]
>> You could also base this on debian:bullseye, which already ships with guix on all archs that guix claims to support as well.

> True, perhaps we can do so in another PR!

I've got a Dockerfile for this. Will PR shortly.

File: guix/Dockerfile (hunk 1 above)
Comment (fanquake) [reply guix/Dockerfile]: Done in 8cb7890890e8a22f15b9cfb32a6cdf8e13dfa124.

**Pull request merged**